nautobot/extras/tests/test_api.py

```diff
@@ -1242,6 +1242,68 @@ def test_run_job_future(self, mock_get_worker_count):
 
         return (response, schedule)  # so subclasses can do additional testing
 
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    @mock.patch("nautobot.extras.api.views.get_worker_count")
+    def test_run_a_job_with_sensitive_variables_for_future(self, mock_get_worker_count):
+        mock_get_worker_count.return_value = 1
+        self.add_permissions("extras.run_job")
+
+        job_model = Job(
+            source="local",
+            module_name="sensitivejob_module",
+            job_class_name="SensitiveJob",
+            grouping="Sensitive Module",
+            name="Job With Sensitive Variable",
+            installed=True,
+            enabled=True,
+            has_sensitive_variables=True,
+        )
+        job_model.validated_save()
+
+        url = self.get_run_url("local/sensitivejob_module/SensitiveJob")
+
+        d = DeviceRole.objects.create(name="role", slug="role")
+        data = {
+            "data": {"var1": "x", "var2": 1, "var3": False, "var4": d.pk},
```

Comment:
Is creating a `DeviceRole` and populating `"data"` with variable values actually needed for this test?